### ISSUE TITLE
Inverted horizontal scrolling detection initialisation is wrong

### ISSUE THREAD

**Issue by tothxa** [widelands/widelands#5542]
**Describe the bug**
The state of inverted scrolling detection is saved in the config file, and so the config option is initialised from there, not from the default value. Because of this, on SDL upgrade from buggy to fixed version the compensation stays turned on, when it shouldn't. https://github.com/widelands/widelands/discussions/5367#discussioncomment-3540461

**To Reproduce**
1. Run Widelands with old/buggy SDL version and exit so that the config is saved, or edit the config file and in the `[mousewheel]` section set `last_autoinvert_x="true"`
2. Start Widelands with fixed or unaffected SDL version
3. Go to About/Technical Info
4. See that horizontal scrolling is wrongly detected as inverted
5. You can test it right there on the tab bar and see that horizontal scrolling is wrong

**Expected behavior**
Compensation for inverted scrolling should be turned off when not needed any more.

**Version:**
 - Widelands Version: 1.1~git25862 (cdb5ccf@master)

**Additional context**
As this is an important bug in a new feature, can it be added to the 1.1 milestone?
